cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/Filters/FilterCollection/FilterConditionInput/index.tsx

```diff
@@ -56,48 +57,53 @@ const FilterConditionInputview: React.FC<IFilterConditionInputProps> = ({
   setFilterToCondition,
   widgetInfo,
 }) => {
-  const existingCondition = filterToCondition[filterID];
+  const existingCondition = filterToCondition.get(filterID);
 
   const [conditionMode, setConditionMode] = React.useState(
-    existingCondition.expression ? FilterConditionMode.Expression : FilterConditionMode.Operator
+    existingCondition.get('expression')
+      ? FilterConditionMode.Expression
+      : FilterConditionMode.Operator
   );
 
   const allWidgetNames = widgetInfo
-    ? Object.values(widgetInfo)
-        .map((info: IWidgetInfo) => info.name)
-        .filter((widgetName) => !isNil(widgetName))
-    : [];
+    ? widgetInfo
+        .valueSeq()
+        .map((info) => fromJS(info).get('name'))
```

Comment:
the values of the info should be an immutable object already

Reply:
Resolved